core/app/models/spree/address/state_validator.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Spree
+  class Address::StateValidator
+    attr_reader :address
+    delegate :state, :state_name, :country, to: :address
+
+    def initialize(address)
+      @address = address
+
+      # normalize state and state_name attributes before
+      # performing the validation
+      Spree::Address.state_normalizer_class.new(address).perform!
```

Comment:
What about passing this class to the `initialize` of this class?

```ruby
def initialize(address, address_normalizer_class = Spree::Address.state_normalizer_class)

# and in the caller
self.class.state_validator_class.new(self, self.class.state_normalizer_class).perform!
```

This way it will still be possible to change the class as an address level preference but we can also have more flexibility if needed for some customization or testing needs.

Reply:
This wasn't intended to be here. I have moved it out into `validate` block in `Spree::Address`. Forgot to remove it from here :|